src/librustc_trans/trans/base.rs

```diff
@@ -777,6 +775,33 @@ pub fn load_if_immediate<'blk, 'tcx>(cx: Block<'blk, 'tcx>,
     return v;
 }
 
+pub fn load_addr(bcx: Block, fatptr: ValueRef, ty: Ty) -> ValueRef {
+    let addr_ptr = expr::get_dataptr(bcx, fatptr);
+    match ty.sty {
+        ty::TyBox(_) | ty::TyRef(..) => {
+            LoadNonNull(bcx, addr_ptr)
+        }
+        _ => Load(bcx, addr_ptr)
```

Comment:
Mmm... that's right, I think.  I was getting confused a bit because this is used to load both fat pointers and dereferences of fat pointers.